frontend/src/components/eventsModal.js

```diff
@@ -0,0 +1,94 @@
+import { Component } from 'react'
+import Router from 'next/router'
+import {
+  Alert,
+  Button,
+  Form,
+  FormGroup,
+  Label,
+  Input,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter
+} from 'reactstrap'
+import { createEvent } from '../utils/api'
+
+class EventsModal extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      alert: ''
+    }
+  }
+
+  addEvent = async () => {
+    const event = {
+      name: this.state.name ? this.state.name : '',
+      date: this.state.date ? this.state.date : '',
+      startTime: this.state.startTime ? this.state.startTime : '',
+      endTime: this.state.endTime ? this.state.endTime : '',
+      location: this.state.location ? this.state.location : '',
+      description: this.state.description ? this.state.description : '',
+      fbLink: this.state.fbLink ? this.state.fbLink : ''
+    }
+    const { code } = await createEvent(event)
+    if (code === 500) {
+      this.setState({
+        alert: 'All fields are required.'
+      })
+    } else {
+      Router.push({ pathname: '/eventOverview' })
+    }
+  }
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+
+  handleKeyPress = async event => {
+    if (event.key === 'Enter') {
+      await this.addEvent()
+    }
+  }
+
+  render() {
+    const alert = this.state.alert ? <Alert color="danger">{this.state.alert}</Alert> : null
+    return (
+      <>
+        <Modal isOpen={this.props.isOpen}>
+          <ModalHeader>{this.props.title}</ModalHeader>
+          <ModalBody>
+            {alert}
```

Comment:
get rid of the ternary on 58 and do something like `this.state.alert && {alert}` maybe? seems cleaner to me but is kind of a nit